### PR TITLE
Jsap 10

### PR DESCRIPTION
**Pull request by eleven032**
JSAP 10 branch work
at backend:
- create db folder, with database connection method.
- create migrations folder, with migration action.
- put env variables into .env file.
- update readme file.

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#JSAP-10
+add new packages mysql and mysql-migration into dependency
+create db folder and create connection function in connection.js
+
+### How mysql migration works
+make sure you are at backend folder
+
+```bash
+$ node ./db/migrations/migration.js add migration create_table_users
```
Comment (matecserven) [inline backend/README.md]: create a script for this, also create_table_users should be dynamic, also write documentation for this part and
https://www.npmjs.com/package/mysql-migrations#executing-migrations this part as well

Comment (matecserven) [reply backend/README.md]: Please write documentation for what environment variables are used for mysql

Comment (eleven032) [reply backend/README.md]: I did some change on package.json and readme, please help me check it, ty

File: backend/package.json [hunk 2]
```diff
@@ -5,13 +5,18 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "node app.js"
+    "start": "node app.js",
+    "createTable": "node ./db/migration.js add migration",
+    "addSeed": "node ./db/migration.js add seed",
+    "dbExecuting": "node ./db/migration.js up"
```
Comment (matecserven) [inline backend/package.json]: lets change this to refresh, so we drop all data and update with new ones

File: backend/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#JSAP-10
+add new packages mysql and mysql-migration into dependency
+create db folder and create connection function in connection.js
+
+## How mysql migration works
+make sure you are at backend folder
+
+# Initiate a migration
+```bash
+$ node run createTable -- <your table name>
```
Comment (matecserven) [inline backend/README.md]:
```suggestion
$ npm run createTable -- <your table name>
```

File: backend/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Note: There is a space between '--' and your table name!!!
+
+This command will create a file start with numbers inside migrations folder
+remember this name
+
+'UP' should fill in the mysql query to create table and 'DOWN' is drop table.
+
+# Add seed
+```bash
+$ node run addSeed -- <table name which you created by above command>
```
Comment (matecserven) [inline backend/README.md]:
```suggestion
$ npm run addSeed -- <table name which you created by above command>
```

File: backend/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+```
+Note: There is a space between '--' and your table name!!!
+
+This command will help insert or update the table data.
+'UP' for insert or update sql query, 'down' for remove data query.
+
+# Executing Migrations
+```bash
+$ node run dbExecuting
```
Comment (matecserven) [inline backend/README.md]:
```suggestion
$ npm run dbRefresh 
```

File: backend/package.json (hunk 2 above)
Comment (matecserven) [inline backend/package.json]:
```suggestion
    "dbRefresh": "node ./db/migration.js refresh"
```

File: backend/README.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+#JSAP-10
+add new packages mysql and mysql-migration into dependency
+create db folder and create connection function in connection.js
```
Comment (matecserven) [inline backend/README.md]: We are going to provide connection and migration files, this part is not needed, also don't need to install dependencies on its own, npm install will do that

**Pull request merged**